### PR TITLE
docs: update major release instructions based on v6 release

### LINKED ISSUE #6761: Docs: Update the 'Major Release Steps' docs for what we did to release v6
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

We've done a few things to prep v6 that aren't in the docs yet. They should be documented.

At the very least:

* #6759
* #6760
* Writing & sharing out a blog post
* Creating a Discord channel

### Affected URL(s)

https://typescript-eslint.io/maintenance/releases#major-release-steps

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6761
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

I believe this captures all the steps we took that aren't yet documented. The main added groups are:

1. `1a. Shared Config Changes`: setting up a discussion for proposed config changes
2. `1b. Voluntary Community Testing`: sending issues & PRs to community repos
3. `1c. Post Community Testing Config Touchups`: changing configs based on feedback, if needed

The Node version changes mentioned in https://github.com/typescript-eslint/typescript-eslint/issues/6761#issuecomment-1629793524 are separately mentioned in `DEPENDENCY_VERSION_UPGRADES.mdx`.

💖

### CODE REVIEW

File: docs/maintenance/Releases.mdx [hunk 1]
```diff
@@ -22,23 +20,59 @@ Per [Users > Releases > Major Releases](../users/Releases.mdx#major-releases), w
...
-     - Its publish command should be `npx lerna publish premajor --loglevel=verbose --canary --exact --force-publish --yes --dist-tag rc-v${major}`.
+1. Raise a PR from `v${major}-canary-auto-release` to `main` modifying [`ci.yml` workflow](https://github.com/typescript-eslint/typescript-eslint/blob/main/.github/workflows/ci.yml) and README.md [example: [chore: add auto-canary release for v6](https://github.com/typescript-eslint/typescript-eslint/pull/5883)]:
+   - `ci.yml`:
+     - Under `pushes:` at the beginning of the file, add an `- v${major}` list item.
+     - Add a `publish_canary_version_v${major}` step the same as `publish_canary_version` except:
+       - Add the condition: `if: github.ref == 'refs/heads/v${major}'`.
+       - Its publish command should be `npx nx release publish premajor --loglevel=verbose --canary --exact --force-publish --yes --dist-tag rc-v${major}`.
+   - `README.md`:
+     - Add a link to a `v${major}--typescript-eslint.netlify.app` preview deploy environment on Netlify that you create for the branch.
    - Merge this into `main` once reviewed and rebase the `v${major}` branch.
```
Comment (JoshuaKGoldberg) [inline docs/maintenance/Releases.mdx:31]: @JamesHenry I defer to your expertise on Lerna/nx changes 😄

**Review by bradzacher — DISMISSED** [review]

File: docs/maintenance/Releases.mdx [hunk 2]
```diff
@@ -22,23 +20,59 @@ Per [Users > Releases > Major Releases](../users/Releases.mdx#major-releases), w
...
 
+#### 1a. Shared Config Changes
+
+Major versions are our only real chance to change the values in our stable `recommended*` and `stylistic*` configs.
+In parallel to the general PR flow of the major version:
+
+1. Create a discussion with a table summarizing any proposed rule changes [example: [Changes to configurations for 6.0.0](https://github.com/typescript-eslint/typescript-eslint/discussions/6014)]
```
Comment (JoshuaKGoldberg) [inline docs/maintenance/Releases.mdx:38]: I suppose this also necessitates making a `v${major}` channel on Discord...

File: docs/maintenance/Releases.mdx
Comment (JoshuaKGoldberg) [inline docs/maintenance/Releases.mdx:1]: Note to self: should also add in something about finding `@deprecated` code that can now be removed. If it isn't there already...

File: docs/maintenance/Releases.mdx (hunk 1 above)
Comment (JamesHenry) [reply docs/maintenance/Releases.mdx:35]: Don't see any issues, thanks!

**Review by bradzacher — DISMISSED** [review]

**Pull request merged**